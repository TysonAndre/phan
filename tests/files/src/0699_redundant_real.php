<?php
class Example699 {
    public $p1;
    public function init() {
        $this->p1 = null;
    }

    public function modify() {
        if (!isset($this->p1)) {
            $this->p1 = 'other';  // Doesn't warn because type was undeclared.
        }
    }
}
