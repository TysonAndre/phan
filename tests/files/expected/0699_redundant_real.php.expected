%s:10 PhanTypeMismatchProperty Assigning 'other' to property but \Example699->p1 is null
